File: session4/lecture91/combi.cpp
#include <bits/stdc++.h>
using namespace std;

int n, m;
vector<vector<int>> v_list;

void combi(int start, vector<int> v)
{
  if (v.size() == m)
  {
    v_list.push_back(v);
    return;
  }
  for (int i = start + 1; i < n; i++)
  {
    v.push_back(i);
    combi(i, v);
    v.pop_back();
  }
  return;
}

int main()
{
  cin >> n >> m;
  vector<int> v;
  combi(-1, v);
  return 0;
}

File: session4/lecture91/combi.js
const input = require('fs').readFileSync('/dev/stdin', 'utf8').trim().split('\n');
let line = 0;
const readLine = () => input[line++];

const [n, m] = readLine().split(' ').map(Number);
const vList = [];

function combi(start, v) {
  if (v.length === m) {
    vList.push([...v]);
    return;
  }
  for (let i = start + 1; i < n; i++) {
    v.push(i);
    combi(i, v);
    v.pop();
  }
}

combi(-1, []);
